api/renter_test.go

```diff
@@ -101,63 +216,227 @@ func TestRenterDownloadError(t *testing.T) {
+	filesize := int64(modules.SectorSize)
+	setupDownloadTest(t, filesize, 0, filesize/2, false)
+}
+
+func TestRenterDownloadShortLengthAndOffsetSingleChunk(t *testing.T) {
+	filesize := int64(modules.SectorSize)
+	setupDownloadTest(t, filesize, filesize/4, filesize/2, false)
+}
+
+func TestRenterDownloadShortLengthTwoChunk(t *testing.T) {
+	filesize := int64(modules.SectorSize * 2)
+	setupDownloadTest(t, filesize, 0, int64(float64(filesize)*0.75), false)
+}
+
+func TestRenterDownloadShortLengthThreeChunkInThirdChunk(t *testing.T) {
+	filesize := int64(float64(modules.SectorSize) * 2.7)
+	setupDownloadTest(t, filesize, 0, int64(2.2*float64(modules.SectorSize)), false)
+}
+
+func TestRenterDownloadShortLengthThreeChunkInSecondChunk(t *testing.T) {
+	filesize := int64(float64(modules.SectorSize) * 2.7)
+	setupDownloadTest(t, filesize, 0, int64(1.6*float64(modules.SectorSize)), false)
+}
+
+func TestRenterDownloadShortLengthMultiChunk(t *testing.T) {
+	filesize := int64(modules.SectorSize * 5)
+	setupDownloadTest(t, filesize, 0, int64(float64(filesize)*0.75), false)
+}
+
+func TestRenterDownloadShortLengthAndOffsetTwoChunk(t *testing.T) {
+	filesize := int64(modules.SectorSize * 2)
+	setupDownloadTest(t, filesize, 50, int64(float64(filesize)*0.75), false)
+}
+
+func TestRenterDownloadShortLengthAndOffsetThreeChunkInSecondChunk(t *testing.T) {
+	filesize := int64(modules.SectorSize * 3)
+	setupDownloadTest(t, filesize, 50, int64(float64(filesize)*0.5), false)
+}
+
+func TestRenterDownloadShortLengthAndOffsetThreeChunkInThirdChunk(t *testing.T) {
+	filesize := int64(modules.SectorSize * 3)
+	setupDownloadTest(t, filesize, 50, int64(float64(filesize)*0.75), false)
+}
+
+// TESTS FOR HTTP RESPONSE BODY BACKED DOWNLOADS.
+func TestRenterDownloadHttpRespOffsetSingleChunk(t *testing.T) {
+	setupDownloadTest(t, int64(modules.SectorSize), 40, int64(modules.SectorSize)-40, true)
+}
+
+func TestRenterDownloadHttpRespOffsetTwoChunk(t *testing.T) {
+	filesize := int64(modules.SectorSize) * 2
+	setupDownloadTest(t, filesize, 40, filesize-40, true)
+}
+
+func TestRenterDownloadHttpRespOffsetManyChunks(t *testing.T) {
+	filesize := int64(modules.SectorSize) * 5
+	setupDownloadTest(t, filesize, 40, filesize-40, true)
+}
+
+func TestRenterDownloadHttpRespOffsetAndLengthSingleChunk(t *testing.T) {
+	filesize := int64(modules.SectorSize)
+	setupDownloadTest(t, filesize, 40, 4*filesize/5, true)
+}
+
+func TestRenterDownloadHttpRespOffsetAndLengthTwoChunk(t *testing.T) {
+	filesize := int64(modules.SectorSize) * 2
+	setupDownloadTest(t, filesize, 80, 3*filesize/4, true)
+}
+
+func TestRenterDownloadHttpRespOffsetAndLengthManyChunks(t *testing.T) {
+	filesize := int64(modules.SectorSize) * 5
+	setupDownloadTest(t, filesize, 150, 3*filesize/4, true)
+}
+
+func TestRenterDownloadHttpRespOffsetAndLengthManyChunksSubsetOfChunks(t *testing.T) {
+	filesize := int64(modules.SectorSize) * 5
+	setupDownloadTest(t, filesize, 150, 1*filesize/4, true)
+}
```

Comment:
These should all be converted to a single larger table-driven test. Here's an example:

https://github.com/NebulousLabs/Sia/blob/master/modules/renter/erasure_test.go#L27